java/Cherepnin_Oleh/src/com/zoolatech/lecture4/tasks/_2/Tree.java

```diff
@@ -0,0 +1,88 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class Tree {
+    private Node head;
+
+    public Tree() {
+    }
+
+    public boolean add(int value) {
+        Node newNode = new Node(value);
+
+        if (head == null) {
+            head = newNode;
+            return true;
+        }
+
+        Node current = this.head;
+        do {
+            if (value == current.getValue()) {
+                return false;
+            }
+            if (value < current.getValue()) {
+                if (current.getLeft() == null) {
+                    current.setLeft(newNode);
+                    return true;
+                } else if (current.getLeft() != null) {
+                    current = current.getLeft();
+                    continue;
```

Comment:
you can remove the `continue` statement if you change the second `if` to `else-if`